hibernate-core/src/main/java/org/hibernate/query/internal/QueryParameterBindingImpl.java

```diff
@@ -180,13 +180,13 @@ public void setBindValue(T value, AllowableParameterType<T> clarifiedType) {
 			this.bindType = clarifiedType;
 		}
 
-		if ( ! getTypeConfiguration().getSessionFactory().getJpaMetamodel().getJpaCompliance().isLoadByIdComplianceEnabled() ) {
+//		if ( ! getTypeConfiguration().getSessionFactory().getJpaMetamodel().getJpaCompliance().isLoadByIdComplianceEnabled() ) {
```

Comment:
Revert?

Reply:
Actually, I don't think this should get reverted.  This setting is intended for load-by-id.  Here we are explicitly dealing with Query.

But I should clean-up the commented code